libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1161,25 +1206,53 @@ void WaveClip::ShiftBy(double delta) noexcept
 
 // Bug 2288 allowed overlapping clips.
 // This was a classic fencepost error.
-// We are within the clip if start < t <= end.
+// We are within the clip if start <= t < end.
```

Comment:
You are changing edge case behavior!

I don't know what subtle off-by-one errors may result, or may be corrected.

I looked back in history and found this comment added at commit e5dc8928dcc3a26bbd878d85682e9dbc1fd6f492 which references  https://bugzilla.audacityteam.org/show_bug.cgi?id=2288

```
// Bug 2288 allowed overlapping clips.
// This was a classic fencepost error.
// We are within the clip if start < t <= end.
// Note that BeforeClip and AfterClip must be consistent 
// with this definition.
```
This commit changed < to <= for the test at the right boundary.

I suggest you go to the Bugzilla page, and do the steps to reproduce, to be sure there is no regression.

Reply:
See also later commits 1cc2f3b6fe250c71fc919b5738e726f360d1758d, a58328c3df0e9ace0362c28e93f449dba3322ee0, to understand the ancestry of this function.

Reply:
> I suggest you go to the Bugzilla page, and do the steps to reproduce, to be sure there is no regression.

Good suggestion. I did and the behavior is correct.

You will see in `b742621` that I augmented the collection of `WaveClip` methods for those kind of comparison, in an effort to make clip boundary checks in `WaveTrack` easier to understand. These new methods (and renamed existing methods) have a consistent documentation and the comment you mention is probably not needed anymore.